packages/kbn-optimizer/limits.yml

```diff
@@ -97,6 +97,7 @@ pageLoadAssetSize:
   mapsEms: 26072
   metricsDataAccess: 73287
   ml: 82187
+  mockIdpPlugin: 30000
```

Comment:
Mind if we add a `ci:cloud-deploy` label?  This seems like a gap in the packaging system with a development plugin included in production.  

I can see we're skipping the server load but I'm not sure how the client is going to behave.

Reply:
@jbudz Both plugins are development-only and are not used in production. The only reason they're broken up into two separate plugins is to avoid cyclic TS projects.

Reply:
Can we add devOnly to https://github.com/elastic/kibana/pull/172257/files#diff-04504e6ebadd3cdc5ecb3ff8e4e2aa2d793812f94fbe64f727b2a4b3fcb79a66R6

Reply:
Yes, can we circle back to https://github.com/elastic/kibana/pull/172257#discussion_r1412067320 and try the label?  It looks like the mock-idp plugin is going to be loaded in browsers for production environments

Reply:
@jbudz I've added the label.

Reply:
The login page is available in production, https://kibana-pr-172257.kb.us-west2.gcp.elastic-cloud.com:9243/mock_idp/login.  I'm not comfortable shipping this as is, from a quality perspective.

Offhand I don't know how to resolve this or have any suggestions.  I'm not familiar with this area.  Maybe core or another developer on operations can help?

Reply:
Thanks for verifying this @jbudz - The login selector should not be available in prod. 

I don't quite understand how the plugin gets loaded in though since it should only be loaded when `opts.dev`, `opts.serverless` and `opts.ssl` are enabled:

https://github.com/elastic/kibana/pull/172257/files#diff-46a1f244256005974ea855d8722369e846f448785f92f9dbc62f5fbaa367b5d8L109-L114

Is `opts.dev` enabled for these QA cloud deployments?

Reply:
> Is opts.dev enabled for these QA cloud deployments?

It's not.  We're using the same build script we use for releases, i.e. the ` node scripts/build --release`

Reply:
@pgayvallet Is there any other mechanism in place where Kibana would load in this plugin despite the `dev` CLI arg not being set?

Reply:
@thomheymann I apologize, I missed the ping. We have one CI script (the SO schema check) that loads Kibana while forcing all plugins to be enabled. but I don't think it would be the issue here. 

However, fwiw, I'm not sure the optimizer / bundler ever supported evicting "devOnly" plugins @jbudz? 

Having the plugin disabled in production for sure. Not bundling specific plugins into the distributable? AFAIK this was never implemented. I think the bundler will consider all plugins as production code?

Reply:
Looking at your PR and at  the `@kbn/mock-idp-plugin`, I'm not sure how we got to the conclusion that this plugin is disabled by default and/or not bundled for production?

The plugin discovery system is now plugged on the package list, so it's not because a plugin in under `package` instead of `/src/plugins` or `/x-pack/plugins` that it will be disabled by default. All packages with a `kibana.jsonc` file will be considered as a plugin and included in the plugin list automatically (and then loaded). 

Also IIRC, the bundler will consider all plugins as production code, meaning that they WILL be included (ignoring, I assume, the `devOnly` flag that is supposed to work on packages, not plugins)

@jbudz will have to confirm those assumption of course, this is Operations' territory, not Core's. But if that was the case, the only approach I would think of is to add a configuration to your plugin with a schema forcing it to be disabled for production, like

```ts
        enabled: schema.conditional(
          schema.contextRef('dist'),
          true,
          schema.boolean({ defaultValue: false, validate: (raw) => {
              if(raw !== false) { throw "mock idp plugin cannot be enabled in production"}
          } }),
          schema.boolean({ defaultValue: true })
        ),
```

Reply:
> However, fwiw, I'm not sure the optimizer / bundler ever supported evicting "devOnly" plugins @jbudz?

It's safe to ignore any of my assumptions here.  I haven't really been involved with any of the packaging changes. @mistic do you know?